components/camel-caffeine/src/main/java/org/apache/camel/component/caffeine/resume/CaffeineCache.java

```diff
@@ -57,7 +58,7 @@ public CaffeineCache(Cache<K, Object> cache, long cacheSize) {
 
     @Override
     public boolean contains(K key, Object entry) {
-        assert key != null;
+        ObjectHelper.notNull(key, "key");
```

Comment:
Assert is the correct check here, as this is set internally by the resume code and it should not be null. The assert only adds a layer of check when running in tests as surefire/failsafe will enable native Java assertions by default in those cases.

Reply:
> The assert only adds a layer of check when running in tests as surefire/failsafe will enable native Java assertions by default in those cases.

But this is not a test running under surefire/failsafe control. So the same reasoning applies here as :point_up_2:.

Reply:
I am wrong about this one. `ObjectHelper` is OK here.